R/mlPrediction.R

```diff
@@ -74,54 +74,60 @@ is.jaspMachineLearning <- function(x){
 }
 
 # S3 method to make predictions using the model
-.mlPredictionGetPredictions <- function(model, dataset){
+.mlPredictionGetPredictions <- function(model, dataset) {
   UseMethod(".mlPredictionGetPredictions", model)
 }
-.mlPredictionGetPredictions.lda <- function(model, dataset){
+.mlPredictionGetPredictions.lda <- function(model, dataset) {
   as.character(MASS:::predict.lda(model, newdata = dataset)$class)
 }
-.mlPredictionGetPredictions.gbm <- function(model, dataset){
-  if(inherits(model, "jaspClassification")){
+.mlPredictionGetPredictions.gbm <- function(model, dataset) {
+  if (inherits(model, "jaspClassification")) {
     tmp <- gbm:::predict.gbm(model, newdata = dataset, n.trees = model[["n.trees"]], type = "response")
     as.character(colnames(tmp)[apply(tmp, 1, which.max)])
-  } else if(inherits(model, "jaspRegression")){
+  } else if (inherits(model, "jaspRegression")) {
     as.numeric(gbm:::predict.gbm(model, newdata = dataset, n.trees = model[["n.trees"]], type = "response"))
   }
 }
-.mlPredictionGetPredictions.randomForest <- function(model, dataset){
-  if(inherits(model, "jaspClassification")){
+.mlPredictionGetPredictions.randomForest <- function(model, dataset) {
+  if (inherits(model, "jaspClassification")) {
     as.character(randomForest:::predict.randomForest(model, newdata = dataset))
-  } else if(inherits(model, "jaspRegression")){
+  } else if (inherits(model, "jaspRegression")) {
     as.numeric(randomForest:::predict.randomForest(model, newdata = dataset))
   }
 }
-.mlPredictionGetPredictions.cv.glmnet <- function(model, dataset){
+.mlPredictionGetPredictions.cv.glmnet <- function(model, dataset) {
   as.numeric(glmnet:::predict.cv.glmnet(model, newx = data.matrix(dataset)))
 }
 
 # S3 method to make find out number of observations in training data
-.mlPredictionGetTrainingN <- function(model){
+.mlPredictionGetTrainingN <- function(model) {
   UseMethod(".mlPredictionGetTrainingN", model)
 }
-.mlPredictionGetTrainingN.lda <- function(model){
+.mlPredictionGetTrainingN.lda <- function(model) {
   model[["N"]]
 }
-.mlPredictionGetTrainingN.gbm <- function(model){
+.mlPredictionGetTrainingN.gbm <- function(model) {
   model[["nTrain"]]
 }
-.mlPredictionGetTrainingN.randomForest <- function(model){
+.mlPredictionGetTrainingN.randomForest <- function(model) {
   length(model[["y"]])
 }
-.mlPredictionGetTrainingN.cv.glmnet <- function(model){
+.mlPredictionGetTrainingN.cv.glmnet <- function(model) {
   model[["glmnet.fit"]][["nobs"]]
 }
 
-.mlPredictionReadModel <- function(options){
-  if(options[["file"]] != ""){
-    model <- try({ readRDS(options[["file"]]) })
-    if(!is.jaspMachineLearning(model))
+.load_jaspMLobject <- function(file) {
+  tmp <- new.env()
+  load(file = file, envir = tmp)
+  tmp[[ls(tmp)[1]]]
+}
+
+.mlPredictionReadModel <- function(options) {
+  if (options[["file"]] != "") {
+    model <- try({ .load_jaspMLobject(file = options[["file"]]) })
```

Comment:
yeah so with `readRDS` wouldn't this just be `model <- readRDS(file = ...)`? As in, no need for this part anymore:
```r
tmp <- new.env()
load(file = file, envir = tmp)
tmp[[ls(tmp)[1]]]
```
(the try is a good idea nevertheless because this can probably fail in many ways).